app/com/arpnetworking/metrics/portal/reports/impl/chrome/DevToolsNetworkConfigurationProtocol.java

```diff
@@ -0,0 +1,137 @@
+
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.github.kklisura.cdt.protocol.commands.Fetch;
+import com.github.kklisura.cdt.protocol.commands.Network;
+import com.github.kklisura.cdt.protocol.types.fetch.HeaderEntry;
+import com.github.kklisura.cdt.protocol.types.network.ErrorReason;
+import com.github.kklisura.cdt.protocol.types.network.Request;
+import com.github.kklisura.cdt.protocol.types.network.RequestPattern;
+import com.github.kklisura.cdt.services.ChromeDevToolsService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/* package private */ enum DevToolsNetworkConfigurationProtocol {
+
+    DEPRECATED_NETWORK(DevToolsNetworkConfigurationProtocol::configureWithNetwork),
+    FETCH(DevToolsNetworkConfigurationProtocol::configureWithFetch);
+
+    DevToolsNetworkConfigurationProtocol(final BiConsumer<ChromeDevToolsService, PerOriginConfigs> configure) {
+        _configure = configure;
+    }
+
+    public void configure(final ChromeDevToolsService dts, final PerOriginConfigs originConfigs) {
+        _configure.accept(dts, originConfigs);
+    }
+
+    private final BiConsumer<ChromeDevToolsService, PerOriginConfigs> _configure;
+
+    @SuppressWarnings("deprecation")
+    private static void configureWithNetwork(final ChromeDevToolsService dts, final PerOriginConfigs originConfigs) {
+
+        final Network network = dts.getNetwork();
+        network.setRequestInterception(ImmutableList.of(new RequestPattern()));
+        network.onRequestIntercepted(event -> {
+            final String url = event.getRequest().getUrl();
+            if (!originConfigs.isRequestAllowed(url)) {
+                LOGGER.warn()
+                        .setMessage("rejecting request")
+                        .addData("url", url)
+                        .log();
+                network.continueInterceptedRequest(
+                        event.getInterceptionId(), ErrorReason.ABORTED, null, null, null, null, null, null
+                );
+                return;
+            }
+            final ImmutableMap<String, Object> headers = ImmutableMap.<String, Object>builder()
+                    .putAll(event.getRequest().getHeaders())
+                    .putAll(originConfigs.getAdditionalHeaders(url))
+                    .build();
+            network.continueInterceptedRequest(
+                    event.getInterceptionId(),
+                    null,
+                    null,
+                    url,
+                    event.getRequest().getMethod(),
+                    event.getRequest().getPostData(),
+                    headers,
+                    null
+            );
+        });
+    }
+
+    private static void configureWithFetch(final ChromeDevToolsService dts, final PerOriginConfigs originConfigs) {
```

Comment:
(this comes from https://github.com/ArpNetworking/metrics-portal/blob/6cf37e6eb6ce1ed96420479b29b6dec1acaf0fde/app/com/arpnetworking/metrics/portal/reports/impl/chrome/DevToolsServiceWrapper.java#L103 )